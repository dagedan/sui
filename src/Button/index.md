
## Button

Demo:

```tsx
import React from 'react';
import { Button } from 'sui';

export default () => <Button>sdf老师款到发货考虑好</Button>;
```

More skills for writing demo: https://d.umijs.org/guide/basic#write-component-demo
