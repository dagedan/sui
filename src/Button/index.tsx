import React, { FC } from 'react';
import './index.less';
interface ButtonProps {
  onClick?: React.MouseEventHandler<HTMLElement>;
  type?: 'default' | 'primary' | 'secondary';
  disabled?: boolean;
}
const Button: FC<ButtonProps> = ({ children, onClick, type = 'button' }) => {
  return (
    <div className={'biz_dev_dialog'} onClick={onClick}>
      {children}
    </div>
  );
};
export default Button;
