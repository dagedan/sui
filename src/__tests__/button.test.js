import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../Button';

Enzyme.configure({ adapter: new Adapter() });

describe('Button组件相关', () => {
  it('Button渲染样式正常', () => {
    const wrapper = shallow(<Button />);
    // expect(wrapper).toMatchSnapshot();// 进行快照保存
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
