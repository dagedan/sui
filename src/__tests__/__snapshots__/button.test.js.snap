// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button组件相关 Button渲染样式正常 1`] = `"<div className=\\"biz_dev_dialog\\" onClick={[undefined]} />"`;
